src/Integration/EnvironmentIntegration.php

```diff
@@ -48,8 +48,12 @@ private function updateRuntimeContext(?RuntimeContext $runtimeContext): RuntimeC
         return $runtimeContext;
     }
 
-    private function updateServerOsContext(?OsContext $osContext): OsContext
+    private function updateServerOsContext(?OsContext $osContext): ?OsContext
     {
+        if (!function_exists('php_uname')) {
+            return null;
```

Comment:
This should probably just return `$osContext` since it could contain context from other sources.

And added a backslash to keep CI happy :)

```suggestion
        if (!\function_exists('php_uname')) {
            return $osContext;
```

Reply:
Good call on returning `$osContext` here...I should've done that to begin with. Just committed these changes...thanks!